views/js/qtiCreator/widgets/states/Deleting.js

```diff
@@ -145,7 +145,8 @@ define([
         cols = gridUnits.redistribute(cols);
 
         _.each(cols, function (col) {
-            col.elt.removeClass(`col-${col.units}`).addClass(`col-${col.refactoredUnits}`);
+            let oldClass = col.elt.context.classList.value
+            col.elt.removeClass(oldClass.match(/col-([\d]+)/).input).addClass(`col-${col.refactoredUnits}`);
```

Comment:
Did you check how it is setting refactoredUnits?

Reply:
Yes, I don't think  this is something weird, this is the expected behavior.  :blush:
`cols` are refactored in ` views/js/qtiCreator/helper/gridUnits.js` in lines from 154 to 178.
In my first approach to solution I removed this lines and this solved the problem. But then realized that this happened when there was any resizing in cols (either manual or automatics) so had to find a different approach. 
After that I put back the lines and the resizing occurs as it is supposed to.